lib/cli/cliHost.js

```diff
@@ -0,0 +1,299 @@
+                    }
+                }
+
+                // Move all instances from the deleted host to the current one
+                const newHostname = tools.getHostName();
+                const instances = yield enumInstances(objects);
+                const instancesToRename = instances.filter(i => i.common.host = hostname);
+                for (const instance of instancesToRename) {
+                    yield this.changeInstanceHost(objects, instance, newHostname);
+                }
+
+                // Notify the user that we are done
+                CLI.success.hostDeleted(hostname);
+                return void callback();
+
+            })().catch(err => {
+                CLI.error.unknown(err);
+                return void callback(1);
+            });
+        });
+    }
+
+    /**
+     * Renames the host with the given name to the current one (opposite of `set()`)
+     * @param {any[]} args
+     */
+    rename(args) {
+        const { callback } = this.options;
+
+        /** @type {string} */
+        const oldHostname = args[1];
+        if (!oldHostname) {
+            CLI.error.requiredArgumentMissing('oldHostname', 'host oldHostname');
+            return void callback(34);
+        }
+        this.renameHost(oldHostname, os.hostname());
+    }
+
+    /**
+     * Renames the host with the hostname `oldHostname` to the hostname `newHostname`
+     * @param {string?} oldHostname The hostname to rename from or `undefined` to rename all hosts (single-host mode only!)
+     * @param {string} newHostname The hostname to rename to
+     */
+    renameHost(oldHostname, newHostname) {
+        const { callback, dbConnect, showHelp } = this.options;
+        dbConnect((objects, states, isOffline) => {
+            tools.poorMansAsync(function* () {
+                if (!isOffline) {
+                    CLI.error.cannotChangeRunningSystem();
+                    return void callback(30);
+                }
+
+                const hosts = yield enumHosts(objects);
+                if (hosts.length > 1 && oldHostname === undefined) {
+                    // iobroker host this/self in multihost is not allowed
+                    CLI.error.noHostThisInMultihost();
+                    showHelp();
+                    return void callback(30);
+                }
+
+                // Does another host with the target name exist?
+                if (hosts.find(obj => obj.common.hostname === newHostname)) {
+                    // TODO: Is this the correct behavior?
+                    CLI.error.hostAlreadyExists(newHostname);
+                    return void callback(30);
+                }
+
+                // Remember the new hostname in the system settings
+                const configFileName = tools.getConfigFileName();
+                const configData = JSON.parse(fs.readFileSync(configFileName, 'utf8'));
+                if (configData.system && configData.system.hostname !== newHostname) {
```

Comment:
Should we also pin it in config when we set to os.hostname()? Thin old code only pinned the name if a custom name was used

Reply:
Ok, I read it wrong, so yes the hostname is set by each rename when the old name was gives as parameter ... question is if it makes sense to "persist" the hostname in system config when the "default system hostname" is used

Reply:
What do you mean under "default system hostname" ? E.g. "raspi" ? Or "default" ?

Reply:
I think he means if we should also persist the hostname when the host's actual hostname `os.hostname()` is used as the iobroker hostname. Or should we delete it then from the system config?

Reply:
Exactly. my question was if we should store the name also when os.hostname() is the same. Currently this is done, but it is also the standard behaviour - when no name is set in system config then the os.hostname() is used anyway

Reply:
It is for the case, if you will add second slave. 
Elsewise in this case we must to first to change "default" host to os.hostname() and then add second slave.
Just always use os.hostname()....

Reply:
And what the behavior in this PR?

Reply:
In this PR the behavior is like this:
When the hostname in the system config is different from the one that should be set, it is updated. But I think this might be wrong if we're renaming a host that is not the current one...

Can you explain when the hostname in the system config should be updated? It was not clear from the old spaghetti code.

Reply:
I guess `configData.system.hostname !== newHostname` should actually be `configData.system.hostname === oldHostname`, correct?

Reply:
@GermanBluefox ? How we can continue here?

Reply:
Also ich hebe jetzt 1 Stunde analysiert und die aktuelle Implementierung ist logisch.

Reply:
Also doch nichts mehr ändern und alles so lassen?